programs/token/src/instructions/initialize_multisig.rs

```diff
@@ -0,0 +1,71 @@
+use pinocchio::{
+    account_info::AccountInfo,
+    cpi::slice_invoke_signed,
+    instruction::{AccountMeta, Instruction, Signer},
+    ProgramResult,
+};
+
+extern crate alloc;
+use alloc::vec::Vec;
+
+/// Initialize a new Multisig.
+///
+/// ### Accounts:
+///   0. `[writable]` The multisig account to initialize.
+///   1. `[]` Rent sysvar
+///   2. ..`2+N`. `[]` The N signer accounts, where N is between 1 and 11.
+pub struct InitializeMultisig<'a> {
+    /// Multisig Account.
+    pub multisig: &'a AccountInfo,
+    /// Rent sysvar Account.
+    pub rent_sysvar: &'a AccountInfo,
+    /// Signer Accounts
+    pub multisig_signers: &'a [&'a AccountInfo],
+    /// The number of signers (M) required to validate this multisignature
+    /// account.
+    pub m: u8,
+}
+
+impl InitializeMultisig<'_> {
+    #[inline(always)]
+    pub fn invoke(&self) -> ProgramResult {
+        self.slice_invoke_signed(&[])
+    }
+
+    pub fn slice_invoke_signed(&self, signers: &[Signer]) -> ProgramResult {
+        let &Self {
+            multisig,
+            rent_sysvar,
+            multisig_signers,
+            m,
+        } = self;
+        // Account metadata
+        let mut account_metas = Vec::with_capacity(1 + multisig_signers.len());
+        account_metas.extend_from_slice(&[
+            AccountMeta::writable(multisig.key()),
+            AccountMeta::readonly(rent_sysvar.key()),
+        ]);
+        account_metas.extend(
+            multisig_signers
+                .iter()
+                .map(|a| AccountMeta::readonly(a.key())),
+        );
+
+        // Instruction data layout:
+        // -  [0]: instruction discriminator (1 byte, u8)
+        // -  [1]: m (1 byte, u8)
+        let instruction_data = &[2, m];
+
+        let instruction = Instruction {
+            program_id: &crate::ID,
+            accounts: &account_metas,
+            data: instruction_data,
+        };
```

Comment:
*minor*

If you use the same name for the variable as the field name, you can use a short assignment syntx.
As this method is so small, it is probably fine rename `accounts_metas` to just `accounts`, and `instruction_data` to just `data.
And then you'll have:

```suggestion
        let instruction = Instruction {
            program_id: &crate::ID,
            accounts,
            data,
        };
```

Reply:
yeah i would have done that , but i just tried to be uniform with other methods just followed the same conventions